tests/group_functions/group_broadcast.h

```diff
@@ -131,87 +131,127 @@ void broadcast_sub_group(sycl::queue& queue) {
   sycl::range<D> work_group_range = sycl_cts::util::work_group_range<D>(queue);
 
   // array to return results
-  T res[test_matrix + 1] = {splat_init<T>(0)};
+  T origin_values[test_matrix] = {splat_init<T>(0)};
+  T broadcasted_values[test_matrix] = {splat_init<T>(0)};
   {
-    sycl::buffer<T, 1> res_sycl(res, sycl::range<1>(test_matrix + 1));
+    sycl::buffer<T, 1> origin_values_buf(origin_values,
+                                         sycl::range<1>(test_matrix));
+    sycl::buffer<T, 1> broadcasted_values_buf(broadcasted_values,
+                                              sycl::range<1>(test_matrix));
 
     queue.submit([&](sycl::handler& cgh) {
-      auto res_acc =
-          res_sycl.template get_access<sycl::access::mode::read_write>(cgh);
+      auto origin_values_acc =
+          origin_values_buf.template get_access<sycl::access::mode::read_write>(
+              cgh);
+      auto broadcasted_values_acc =
+          broadcasted_values_buf
+              .template get_access<sycl::access::mode::read_write>(cgh);
 
       sycl::nd_range<D> executionRange(work_group_range, work_group_range);
+      // Values computed in a kernel depend on global linear id. We need to make
+      // sure that there are no overflows
+      REQUIRE(executionRange.get_global_range().size() <
+              std::numeric_limits<size_t>::max() / 100);
 
       cgh.parallel_for<broadcast_sub_group_kernel<
           D, T>>(executionRange, [=](sycl::nd_item<D> item) {
         sycl::sub_group sub_group = item.get_sub_group();
 
+        // Each work-item computes a unique value
+        T value_to_broadcast(splat_init<T>(static_cast<size_t>(
+            item.get_global_linear_id() * 100 + sub_group.get_local_id())));
         T local_var(splat_init<T>(0));
 
+        // To simplify the test, we are only checking the first sub-group
         if (sub_group.get_group_id()[0] == 0) {
-          // find local id of last group item
-          sycl::id<1> last_item = sub_group.get_local_range();
-          --last_item[0];
-
-          // broadcast from the first workitem
-          local_var = splat_init<T>(item.get_global_linear_id() + 1);
+          // Find local id of first, last and some third sub-group item in
```

Comment:
```suggestion
          // Find local id of first, last and some third sub-group id in
```
To be less confusing, since `sycl::item` is also a thing.
I had to look at the spec to refresh my memory on sub-groups.
It might be clearer to add a comment before `first_id` to say it is the sub-group leader or `sub_group::leader()`?

Reply:
Reworded the comment in 2952e4e0fee35fe8d3d4ff9a23b5e48abd4ff311